### PR TITLE
[Woo POS] Entry point won't appear unless restart of the app

### LINKED ISSUE #12740: [Woo POS] Entry point won't appear unless I restart the app
When I install the app and login, I don't see the POS entry point, which is expected. However, when I complete IPP Onboarding, the POS entry point is still hidden. I have to restart the app for it too appear. I experienced this behavior when I tested the build for Woo DM on WooDM Swag store.

Comment from Andrei:
`Yep, I reproduced this too. I briefly checked the code, and it looks like we fetch FFs only when the app starts, and as the app’s start happens in the case of login flow when a user is not logged in, I’d assume that there is no way to white list based on the account name`

### PR DESCRIPTION
**Pull request by kidinov**
Closes: #12740

### Description

We fetch remote FF only on the app starts, so the cases when:
* FF rules contain account restrictions
* FF rules contain site restrictions
are not handled properly on our side and a user has to restart the app.

The change is here to fetch FF on the site change to address that

### Steps to reproduce

* On the clean build login with an account that whitelisted for POS
* Notice that POS is not shown in the MORE menu unless you restart the app

### The tests that have been performed

^

- [x] I have considered if this change warrants release notes and have added them to `RELEASE-NOTES.txt` if necessary. Use the "[Internal]" label for non-user-facing changes.

## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:

- [x] The PR is small and has a clear, single focus, or a valid explanation is provided in the description. If needed, please request to split it into smaller PRs.
- [x] Ensure Adequate Unit Test Coverage: The changes are reasonably covered by unit tests or an explanation is provided in the PR description.
- [x] Manual Testing: The author listed all the tests they ran, including smoke tests when needed (e.g., for refactorings). The reviewer confirmed that the PR works as expected on big (tablet) and small (phone) in case of UI changes, and no regressions are added.

### CODE REVIEW

**Comment by dangermattic** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true" data-kind="Warning">
          1 Warning
      </th>
     </tr>
  </thead>
  <tbody>
    <tr>
      <td>:warning:</td>
      <td data-sticky="false">This PR is assigned to the milestone <a href="https://github.com/woocommerce/woocommerce-android/milestone/242">21.0</a>. This milestone is due in less than 2 days.<br />
Please make sure to get it merged by then or assign it to a milestone with a later deadline.</td>
    </tr>
  </tbody>
</table>

<p align="right" data-meta="generated_by_pr-check">
  Generated by :no_entry_sign: <a href="https://danger.systems/">Danger</a>
</p>

File: WooCommerce/src/main/kotlin/com/woocommerce/android/SiteObserver.kt [hunk 1]
```diff
@@ -46,6 +51,8 @@ class SiteObserver @Inject constructor(
                     launch { fetchOrderStatusOptions(site) }
 
                     launch { sendSiteDataToWearable(site) }
+
+                    launch { fetchRemoteFeatureFlags() }
```
Comment (AnirudhBhat) [inline WooCommerce/src/main/kotlin/com/woocommerce/android/SiteObserver.kt:59]: I think we don't have unit test for this class. We could add one and just test that the `fetchAndCacheFeatureFlags` is called when there is change in selected site to begin with. Definitely not a blocking comment but something to consider either in this or in a separate PR.

**Review by AnirudhBhat — APPROVED** [review]
Thanks for working on this @kidinov. I've just left one non-blocking comment.

**Comment by kidinov** [comment]
@ThomazFB 👋

 Maybe you can point to someone to take a look into this, as we have no expertise in remote FF at all

File: WooCommerce/src/main/kotlin/com/woocommerce/android/SiteObserver.kt (hunk 1 above)
Comment (kidinov) [reply WooCommerce/src/main/kotlin/com/woocommerce/android/SiteObserver.kt:59]: Good call

Added here 9096b55f5680f0103c2b5d3b2a070f700e82a26c

**Review by samiuelson — APPROVED** [review]
LGTM! :shipit:

**Pull request merged**